File: DMOJ/ccc11j2.java
import java.util.Scanner;

public class WhoHasSeenTheWind {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int h = input.nextInt();
        int m = input.nextInt();
        for (int t = 1; t <= m; t++) {
            double a = -6 * (Math.pow(t, 4)) + h * (Math.pow(t, 3)) + 2 * (Math.pow(t, 2)) + t;
            if (a <= 0) {
                System.out.println("The balloon first touches ground at hour:");
                System.out.println(t);
                System.exit(0);
            }
        }
        System.out.println("The balloon does not touch ground in the given time.");
    }
}

File: DMOJ/ccc11j2.py
h = int(input())
waitTime = int(input())
touchesGround = False
for t in range(1, waitTime):
    height = -6 * t * t *t + h*t*t + 2*t + 1
    if height <= 0:
        print(f"The balloon first touches ground at hour:\n {t}")
        touchesGround = True
        break

if not touchesGround:
    print("The balloon does not touch ground in the given time.")
